### PR TITLE
Arse Nath tweaks

### PR DESCRIPTION
**Pull request by Superlagg**
## About the PR 

Reworks Rathens' Secret to be little bit fairer and a lot bit bloodier

First, it'll sever their butt. As usual, but now it uses a slightly fancier random-throw proc, so it'll fly off in more random directions.

After that, it'll try severing a single limb. It did this before, but now it has a few considerations:
- It wont try severing a limb if there isn't a second one of them attached. Unless there's no choice, then it'll pop it right off.
- Non-light robolimbs, bear arms, and shambler appendages won't be removed.
- Light robolimbs, wendigo arms, wolf paws, and stone limbs have a 50% chance to resist severing
-- Though if a limb isn't severed, it'll try again.
- Each check of a limb has a 25% chance to not be chosen, unless its the last one on the list.
- Tails have a 25% chance per missing limb to be severed along with a limb.
-- So you'll only lose your tail if you're also missing a limb.

And in addition to all that, you also spray gibs out of your butt, even if you had a butt when the spell was cast. This is because I felt that making the limb removal less (potentially) debilitating needed something to balance it out. So, now it makes the floors really slippery, something that helps the wizard, but is also countered with adequate footwear.

Also added some flavor text for when their arms resist removal.

Also added limb kind flags to the limbs. Currently, most of them don't do much, but it should help cut down on needing to typecheck a heckload of limbs if you're just looking for a specific thing common to a set of limbs.

## Why's this needed? 

I misjudged how unbelievably annoying it was for tailhavers going up against wizards. Figured I'd may as well do the other things too.

## Changelog

```changelog
(u)Superlagg:
(+)NT struck a deal with the Wizard Federation to tone down Rathens' Secret regarding tails and cyberlimbs, in exchange for shooting a lot more blood and guts out of peoples' butts.
(+)Rathens' Secret will only sever tails if their owner is missing limbs.
(+)It'll also fail to sever most cyberlimbs and exotic appendages.
```

### CODE REVIEW

**Comment by Superlagg** [comment]
So the random gib-flinging didnt work right, since there were multiple different definitions of streak() on a bunch of separate types. I combined them into two procs, one that lives in Cleanables and works based on the calling cleanable's type, and one that lives in Items, which accepts a streak decal thing as an arg. Its not particularly pretty or elegant, but it got around me needing to duplicate a bunch of code. And it works!

**Comment by Superlagg** [comment]
Also regarding the new(ish) random gib pathing thing:
- It works by picking a turf along the edge of a set-size square (or specific side of that square), drawing a line to it, then step-looping its way to it. 
  - Bit more complicated than before, where it just picked a direction and scooted that-a-ways till it hit something or ran out of moves. However `getline()` assures me that the proc is "ultrafast" so it shouldn't be a problem.
- Changed the weighted random number of tiles they scoot to a random 1-6 tiles, with a 10% chance to instead travel around 30 tiles.
  - This is so that, occasionally, a gib'll end up somewhere funky, either making the janitor's job a bit more hide-and-seeky, or making it more obvious that someone exploded somewhere in this area.
  - In testing, these long-shot gibs tended to end up in a distant corner of a room or the exit from maint. Finding some gibs just as I entered maint, then nothing until I wandered in deeper, kinda gave the area kind of a dangerous-intrigue vibe.
- Gibs'll also try to move their full distance, even if its blocked.
  - This means gibs should scoot along a wall if they're instructed to move somewhere behind the wall. Looks kinda neat, honestly.
  - Though this does look kinda weird for the full-distance gibs if they end up in a corner, just kind of jiggling in place until it runs out of moves. It's pretty brief though, it should only last around 3 seconds, and its rare enough (and typically secondary to something more eye-catching, like the person gibbing) that it shouldn't be a problem.
  - Gibs also have a randomized max number of decals it can make, so the full-distance gibs'll only spray decals for only part of their journey.

**Comment by Sovexe** [comment]
I can't actually remember without double checking but...

Is this going to spam a thousand thrown and hit messages lagging the entire server and crashing clients whenever it's used in a large crowd as hundreds gib objects are unpooled and launched to target turfs all at once?

**Comment by Superlagg** [comment]
> I can't actually remember without double checking but...
> 
> Is this going to spam a thousand thrown and hit messages lagging the entire server and crashing clients whenever it's used in a large crowd as hundreds of gib objects are unpooled and launched to target turfs all at once?

No. At least no more than usual. The gib objects don't do anything when they collide with something, aside from keep trying to move, so there'd be no additional browserspam. There would be about 5 more gibs spawned per person, in addition to the blood decals they'd make and the bloodpools those'd end up making.  So it'd be more server intensive than before, but it shouldn't be *that* much more.

File: _std/defines/item.dm [hunk 1]
```diff
@@ -103,3 +103,39 @@
...
+/// Limb is made of stone
+#define LIMB_STONE    (1<<12)
+/// Limb typically belongs to a vicious bear
+#define LIMB_BEAR     (1<<13)
+/// Limb typically belongs to a wendigo
+#define LIMB_WENDIGO  (1<<14)
+/// Limb typically belongs to a large angry dog
+#define LIMB_WOLF     (1<<15)
+/// Limb is kinda boney
+#define LIMB_SKELLY   (1<<16)
```
Comment (Adharainspace) [inline _std/defines/item.dm:141]: i would appreciate if you added an ```is_mutant_limb(x) HAS_FLAG(x, LIMB_MUTANT)``` macro check for all of these.

also, keep in mind that doing it like this means that only 5 more limb types can be added before we have to figure out a new way to handle / cut down on them, due to bitflags having a maximum size of 1 << 23 (due to byond using floats which have a limit of 24 digits total)

**Review by Adharainspace — CHANGES_REQUESTED** [review]

File: _std/defines/item.dm (hunk 1 above)
Comment (Superlagg) [reply _std/defines/item.dm:141]: I added the macros!

**Review by pali6 — APPROVED** [review]

**Comment by pali6** [comment]
I scrolled through the code and nothing seemed extremely terrible. yolo

**Pull request merged**